device/manager.go

```diff
@@ -149,41 +148,32 @@ func (m *manager) Connect(response http.ResponseWriter, request *http.Request, r
 		return nil, ErrorMissingDeviceNameContext
 	}
 
-	var (
-		partnerIDs                   []string
-		satClientID                  string
-		trust                        = secure.Untrusted
-		secureContext, securePresent = handler.FromContext(request.Context())
-	)
+	metadata, providedDeviceMetadata := GetDeviceMetadata(ctx)
 
-	if securePresent {
-		partnerIDs = secureContext.PartnerIDs
-		satClientID = secureContext.SatClientID
-		trust = secureContext.Trust
+	if !providedDeviceMetadata {
+		metadata = NewDeviceMetadata()
 	}
 
 	cvy, cvyErr := m.conveyTranslator.FromHeader(request.Header)
 	d := newDevice(deviceOptions{
-		ID:          id,
-		C:           cvy,
-		Compliance:  convey.GetCompliance(cvyErr),
-		QueueSize:   m.deviceMessageQueueSize,
-		PartnerIDs:  partnerIDs,
-		SatClientID: satClientID,
-		Trust:       trust,
-		Logger:      m.logger,
+		ID:         id,
+		C:          cvy,
+		Compliance: convey.GetCompliance(cvyErr),
+		QueueSize:  m.deviceMessageQueueSize,
+		Metadata:   metadata,
+		Logger:     m.logger,
 	})
 
+	if !providedDeviceMetadata {
+		d.errorLog.Log(logging.MessageKey(), "missing security information")
```

Comment:
Can't this go in the if statement above? They're both checking the same thing.

Reply:
I think this is done so we can use the device's logger

Reply:
I'm talking about the if statement, not the log message.  We check the same parameter (`!providedDeviceMetadata`) in two separate if statements that each have one line in them:
https://github.com/xmidt-org/webpa-common/pull/447/files#diff-540e973a8df442549b241f323503fb38R167
and
https://github.com/xmidt-org/webpa-common/pull/447/files#diff-540e973a8df442549b241f323503fb38R153

Reply:
The second if condition was actually not sufficient for its purspose. I changed it https://github.com/xmidt-org/webpa-common/pull/447/commits/42263b806f0c2c8e6c75741921ff21b8fa418a1b

Reply:
But I'm going through to ensure it does what I think it should